tests/testsrunner.ts

```diff
@@ -0,0 +1,69 @@
+import { AppConfig } from './appconfig'
+import type { BaseTests } from './contracts/basetests'
+import { RaffleTests } from './contracts/raffletests'
+import { VotingTests } from './contracts/votingtests'
+
+export class TestsRunner {
+  private readonly tests: Map<string, BaseTests> = new Map<string, BaseTests>()
+  private readonly config: AppConfig = new AppConfig()
+
+  constructor() {
+    this.config.parseConfig()
+    this.tests.set('voting', new VotingTests(this.config))
+    this.tests.set('raffle', new RaffleTests(this.config))
+  }
+
+  public async runTests(): Promise<boolean> {
+    const specificTests = this.getSpecificTests()
+    let testsToRun = specificTests.length
+    let succeededTests = 0
+    for (const testName of specificTests) {
+      const test = this.tests.get(testName)
+      if (test === undefined) {
+        console.error(`Could not find test ${testName}`)
+        --testsToRun
```

Comment:
```suggestion
        throw new Error(`Could not find test ${testName}`)
```

Reply:
That would cause the app to exit which I don't want. Let it still continue with other found tests.

Reply:
I think the problem inherently comes from having a custom test runner.
Have you considered something like [TestRunner](https://nodejs.org/api/test.html)?
It would help run tests per file, this gives the person running tests ability to control what someone runs.
WDYT?

Reply:
I have not. Will take a look at it later. The implemented test runner is very simple and I'm not sure there's any need to change it or make it more complex :)

Reply:
I think you are reinventing the wheel. 
Eg I can't run a single test with the current implementation, I have to run all or change the code. 
Test runner comes bundled with node and is well documented and tested. I don't think that is complexity as compared to having to reference a custom impl. The changes requested are not major, you just have to wrap the calls eg:
```js
test('Raffle Contract', async (t) => {
  await t.test('init' , (t) => {
    assert.strictEqual(1, 1);
  });

  await t.test('create_proposal;', (t) => {
    assert.strictEqual(2, 2);
  });
}); 
``` 
Now:
- I can run each file alone.
- The test runner manages what fails etc.
- The test runner can output other file formats if needed.

If you are not into working on it, I can do it because I don't feel ok with the current impl.

Reply:
I think you misunderstand the idea of this tool. Its purpose is not to run each of the method alone - for this you have the rust unit tests. Idea of this tool is to test the specific scenario. Currently implemented scenarios are based on the tests Eloy did manually to test the contracts (https://github.com/eigerco/nebula/issues/5). 
And yes if you want to test different scenario - you have to implement another file. Treat the single test file as an integration test case. Actually it might be a good idea to change the test file names as currently they don't tell what is actually being tested.

Reply:
> Its purpose is not to run each of the method alone

I am not suggesting any method to be run alone:
```js
public async run(): Promise<boolean> {
    let result = await this.createAccounts()
    if (!result) return false
    result = await this.deployContract('raffle')
    // ....
}
  ```
  becomes
  ``` js
test('Raffle Contract', async (t) => {
    const contract = new Contract();
    await t.test('createAccounts' , async (t) => {
        await contract.createAccounts()
    });

    await t.test('deployContract', async (t) => {
      await contract.deployContract('raffle')
    });
    .....
});
```
Everything is run the same, but we get a ready test API that is documented and extensible, and we don't have to manage it

Reply:
There's nothing more to manage than in the example you've posted. You still have to write at least the same amount of code. I don't see a value added here.

Reply:
I think I disagree with the notion that it adds no value.
First of all we wouldn't have the `testrunner` file, secondly we don't need to declare a new contract in code, just add the file and it runs.
We plan to have up to 10 contracts, I think I see a lot of value in this.
I can do a PR with the changes I am recommending if need be.

Reply:
Please see https://github.com/eigerco/nebula/pull/47